File: web/utilts/userRoles.ts
import {SimpleUserFragment} from "@/generated/graphql";
import {isUniqueArray} from "@/utilts/arrayUtils";

enum UserRoles {
    ADMIN='ROLE_ADMIN',
    TENANT_OWNER='ROLE_TENANT_OWNER',
    TENANT_ASSIGNED='ROLE_TENANT_ASSIGNED',
    RENTER='ROLE_RENTER'
}

export function isGranted(user: SimpleUserFragment|null, roles: Array<UserRoles>) {
    return !isUniqueArray([...(user?.roles ?? []), ...roles]);
}

export default UserRoles;

File: src/main/kotlin/de/immowealth/entity/UserRoles.kt
package de.immowealth.entity

/**
 * All user roles
 */
class UserRoles {

    companion object {
        /**
         * Admin role
         */
        final val ADMIN = "ROLE_ADMIN";

        /**
         * Tenant owner
         */
        final val TENANT_OWNER = "ROLE_TENANT_OWNER";

        /**
         * Tenant assigned
         */
        final val TENANT_ASSIGNED = "ROLE_TENANT_ASSIGNED"

        final val RENTER = "ROLE_RENTER"
    }
}